### ISSUE TITLE
Trying doing static_quantization with Pytorch 2.1.0 torch.ao.quantization

### ISSUE THREAD

**Issue by RonaldYuren** [pytorch/pytorch#120705]
## Issue description

Using torch.ao.quantization to do Post-training Static Quantization , didn't show error but it's not change the model layer to quantized layer(ex:  nn.Conv2d is still Conv2d after running my code)
Please Help Me with this problem!!!!!   Thank you!

############# Code example

def static_quantize(weight_path, qconfig_mapping):
    weights, imgsz = opt.weights, opt.img_size

    # load Pruned_model
    # 
    device = select_device(opt.device)
    device = torch.device("cpu")
    # Load model
    pruned_model = attempt_load(weights, map_location=device)  # load FP32 model (after pruning from other code)
    
    stride = int(pruned_model.stride.max())  # model stride
    imgsz = check_img_size(imgsz, s=stride)  # check img_size
    pruned_model.eval()
    if weight_path != '1234':  # get pruned_model's weights if path is not default string"1234" 
        pruned_model.load_state_dict(torch.load(weight_path),False)   #load pruned_model.pt

    #  "input"  "output":  Add  QuantStub , DeQuantStub
    
    pruned_model = QuantStub()(pruned_model)
    pruned_model = DeQuantStub()(pruned_model)

    # 
    pruned_model = prepare(pruned_model, qconfig_mapping)
    quantized_model = convert(pruned_model, inplace=False)

    # save model's weights
    torch.save(pruned_model.state_dict(), "static_quantized_original.pt") 
    print("Static Quantization Success!!!!")    
    
    return quantized_model

if __name__ == "__main__":
    device = select_device(opt.device)
    
    
    # create static_quantization_config
    qconfig = torch.ao.quantization.get_default_qconfig('fbgemm')
    qconfig_dict = {'': qconfig}
    qconfig_mapping = QConfigMapping.from_dict(qconfig_dict)

    # operate quantization_function
    quantized_model = static_quantize(weight_path, qconfig_mapping)
    static_quant_weights_path = 'static_quantized_weights.pt'
    torch.save({'model': quantized_model.state_dict()}, static_quant_weights_path)

```

- PyTorch or Caffe2: Pytorch
- How you installed PyTorch (conda, pip, source): conda
- Build command you used (if compiling from source):
- OS: Ubuntu18.04
- PyTorch version:2.1.0
- Python version:3.8.13
- CUDA/cuDNN version:11.6
- GPU models and configuration:
- GCC version (if compiling from source):GCC 7.5.0
- CMake version:
- Versions of any other relevant libraries:

**Comment by supriyar** [comment]
What are the contents of `pruned_model` originally and after you call `prepare` and `convert`?

Can you try with a simpler example, say with one Conv2d layer and see if that works for you?

**Comment by RonaldYuren** [comment]
> What are the contents of `pruned_model` originally and after you call `prepare` and `convert`?
> 
> Can you try with a simpler example, say with one Conv2d layer and see if that works for you?

(1) My Pruned_model is a YOLO model with pruned 1*1 Conv2d layer(**didn't change the model's architecture**)；or just change the activation function 
(2) I have created a simple_Model with only one Conv2d layer , and Use the code below to check whether it work:

################
def check_quantization(model, output_file):
    num_modules_to_print = 5
    with open(output_file, 'w') as f:
        num_printed_modules = 0
        for name, module in model.named_modules():
            if isinstance(module, nn.Conv2d):
                f.write(f"Module {name} is a Conv2d layer.\n")
                f.write(f"Quantized: {isinstance(module, nnq.Conv2d)}\n\n")
            elif isinstance(module, nn.Linear):
                f.write(f"Module {name} is a Linear layer.\n")
                f.write(f"Quantized: {isinstance(module, nnq.Linear)}\n\n")
            # Add more checks for other quantized layers if needed

            # Print some example layers for inspection
            if num_printed_modules < num_modules_to_print:
                f.write(f"+++++++++Module {name}:\n")
                f.write(f"+++++++++{module}\n\n")
                num_printed_modules += 1
######################


**But I always got the outcome below:**

+++++++++Module :
+++++++++SimpleModel(
  (conv): Conv2d(3, 16, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
)

Module conv is a Conv2d layer.
Quantized: False
